### PR TITLE
Core: Add create(..) method to UpdateTableRequest

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/catalog/TableCommit.java [hunk 1]
```diff
@@ -26,6 +26,12 @@
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.immutables.value.Value;
 
+/**
+ * This represents a commit to be applied for a single table with {@link UpdateRequirement}s to be
+ * validated and {@link MetadataUpdate}s that have been applied. The {@link UpdateRequirement}s and
+ * {@link MetadataUpdate}s can be derived from table's base and updated {@link TableMetadata} when
+ * using {@link TableCommit#create(TableIdentifier, TableMetadata, TableMetadata)}.
+ */
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/catalog/TableCommit.java:34]: Can you also add Javadoc for the `create` method? It has requirements for the `TableMetadata` objects that should be noted.

**Review by rdblue — APPROVED** [review]
Looks good overall!

File: core/src/main/java/org/apache/iceberg/catalog/TableCommit.java (hunk 1 above)
Comment (nastra) [reply core/src/main/java/org/apache/iceberg/catalog/TableCommit.java:34]: I've added the Javadoc to make this clearer when using the `create` method

**Comment by nastra** [comment]
Thanks for reviewing @rdblue, I've added the Javadoc to make things clearer

**Pull request merged**